crates/sui-graphql-rpc/src/mutation.rs

```diff
@@ -86,13 +95,40 @@ impl Mutation {
             .map_err(|e| Error::Internal(format!("Unable to execute transaction: {e}")))
             .extend()?;
 
+        let raw_effects: NativeTransactionEffects = bcs::from_bytes(&result.raw_effects)
+            .map_err(|e| Error::Internal(format!("Unable to deserialize transaction effects: {e}")))
+            .extend()?;
+        let sender_signed_data: SenderSignedData = bcs::from_bytes(&result.raw_transaction)
+            .map_err(|e| Error::Internal(format!("Unable to deserialize transaction data: {e}")))
+            .extend()?;
+
+        let events = result
+            .events
+            .ok_or(Error::Internal(
+                "No events are returned from transaction execution".to_string(),
+            ))?
+            .data
+            .into_iter()
+            .map(|e| NativeEvent {
+                package_id: e.package_id,
+                transaction_module: e.transaction_module,
+                sender: e.sender,
+                type_: e.type_,
+                contents: e.bcs,
+            })
+            .collect();
+
         Ok(ExecutionResult {
             errors: if result.errors.is_empty() {
                 None
             } else {
                 Some(result.errors)
             },
-            digest: result.digest.to_string(),
+            effects: TransactionBlockEffects::Executed {
+                tx_data: sender_signed_data,
+                native: raw_effects,
```

Comment:
nit (optional): rename `sender_signed_data` and `raw_effects` to make use of field initializer punning:

```suggestion
                tx_data,
                native,
```